### PR TITLE
Build OS and OSD 1.4.0, remove windows for 1.3.7

### PR DESCRIPTION
**Pull request by gaiksaya**
Signed-off-by: Sayali Gaikawad <gaiksaya@amazon.com>

### Description
In order to unblock CI for plugins that have already upgraded and are in process of upgrading their 1.x branch to 1.4.0, we need to build 1.4.0 artifacts

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by peterzhuamazon — CHANGES_REQUESTED** [review]
Can you remove windows from 1.3.7, and add macos back for 1.4.0?

**Comment by gaiksaya** [comment]
> Can you remove windows from 1.3.7, and add macos back for 1.4.0?

Dont we need windows for 1.3.7?

File: jenkins/check-for-build.jenkinsfile [hunk 1]
```diff
@@ -16,6 +16,8 @@ pipeline {
         parameterizedCron '''
             H 1 * * * %INPUT_MANIFEST=1.3.7/opensearch-1.3.7.yml;TARGET_JOB_NAME=distribution-build-opensearch;BUILD_PLATFORM=linux macos windows
             H 1 * * * %INPUT_MANIFEST=1.3.7/opensearch-dashboards-1.3.7.yml;TARGET_JOB_NAME=distribution-build-opensearch-dashboards;BUILD_PLATFORM=linux
+            H 1 * * * %INPUT_MANIFEST=1.4.0/opensearch-1.4.0.yml;TARGET_JOB_NAME=distribution-build-opensearch;BUILD_PLATFORM=linux
```
Comment (zelinh) [inline jenkins/check-for-build.jenkinsfile:19]: We could also have `macos` for the platform right.

Comment (gaiksaya) [reply jenkins/check-for-build.jenkinsfile]: Added

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**